In [2]:
import pandas as pd

In [23]:
# Cargamos nuestra base de datos

tortilla = pd.read_csv('tortilla_prices.csv')
tortilla

State            City  Year  Month  Day  \
0            Aguascalientes  Aguascalientes  2007      1   10   
1           Baja California        Mexicali  2007      1   10   
2           Baja California         Tijuana  2007      1   10   
3       Baja California Sur          La Paz  2007      1   10   
4                  Campeche        Campeche  2007      1   10   
...                     ...             ...   ...    ...  ...   
284389             Veracruz   Coatzacoalcos  2024      7    8   
284390             Veracruz        Veracruz  2024      7    8   
284391             Veracruz          Xalapa  2024      7    8   
284392              Yucatán          Mérida  2024      7    8   
284393            Zacatecas       Zacatecas  2024      7    8   

               Store type  Price per kilogram  
0       Mom and Pop Store                9.90  
1       Mom and Pop Store                 NaN  
2       Mom and Pop Store               10.00  
3       Mom and Pop Store               10.00  
4       Mom and Pop Store               10.00  
...                   ...                 ...  
284389   Big Retail Store               12.50  
284390   Big Retail Store               13.60  
284391   Big Retail Store               14.33  
284392   Big Retail Store               13.40  
284393   Big Retail Store               13.53  

[284394 rows x 7 columns]

In [17]:
pd.set_option('display.max_columns',None)

In [24]:
# Información del precio de la tortilla en México de 2007 al 2024

tortilla.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 284394 entries, 0 to 284393
Data columns (total 7 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   State               284394 non-null  object 
 1   City                284394 non-null  object 
 2   Year                284394 non-null  int64  
 3   Month               284394 non-null  int64  
 4   Day                 284394 non-null  int64  
 5   Store type          284394 non-null  object 
 6   Price per kilogram  278004 non-null  float64
dtypes: float64(1), int64(3), object(3)
memory usage: 15.2+ MB


"Price per kilogram" es la unica columna que contiene valores nulos

In [25]:
#Valores unicos por cada columna

tortilla.nunique()



State                   32
City                    56
Year                    18
Month                   12
Day                     31
Store type               2
Price per kilogram    1262
dtype: int64

La columna "Price por kilograma" contiene 1262 valores unicos.


In [26]:
# Mostramos el nombre de los valores únicos por columna

for name in df.columns.to_list():
    print(f'La columna: {name} contien los elementos {tortilla[name].unique()}')

La columna: State contien los elementos ['Aguascalientes' 'Baja\xa0California' 'Baja\xa0California\xa0Sur'
 'Campeche' 'Coahuila' 'Colima' 'Chiapas' 'Chihuahua' 'D.F.' 'Durango'
 'Guanajuato' 'Guerrero' 'Hidalgo' 'Jalisco' 'Edo.\xa0México' 'Michoacán'
 'Morelos' 'Nayarit' 'Nuevo\xa0León' 'Oaxaca' 'Puebla' 'Querétaro'
 'Quintana\xa0Roo' 'San\xa0Luis\xa0Potosí' 'Sinaloa' 'Sonora' 'Tabasco'
 'Tamaulipas' 'Tlaxcala' 'Veracruz' 'Yucatán' 'Zacatecas']
La columna: City contien los elementos ['Aguascalientes' 'Mexicali' 'Tijuana' 'La\xa0Paz' 'Campeche'
 'Piedras\xa0Negras' 'Saltillo' 'Colima' 'Tapachula' 'Tuxtla\xa0Gutiérrez'
 'Chihuahua' 'Cd.\xa0Juárez' 'D.F.' 'ZM\xa0D.F.' 'Durango'
 'Gómez\xa0Palacio' 'Irapuato' 'León' 'Acapulco' 'Chilpancingo' 'Pachuca'
 'Guadalajara' 'ZM\xa0Guadalajara' 'Toluca' 'Morelia' 'Cuernavaca' 'Tepic'
 'Monterrey' 'ZM\xa0Monterrey' 'Oaxaca' 'Puebla' 'ZM\xa0Puebla'
 'Querétaro' 'Cancún' 'Chetumal' 'San\xa0Luis\xa0Potosí' 'Culiacán'
 'Cd.\xa0Obregón' 'Hermosillo' 'No

In [21]:
# Obtenemos los valores unicos de la columna price per kilogram

unique_price = tortilla['Price per kilogram'].unique()
unique_price

array([ 9.9 ,   nan, 10.  , ..., 18.3 , 16.72, 19.97])

In [29]:
# Eiminamos filas con valores 'nan'

tortilla_clean = tortilla.dropna(subset=['Price per kilogram'])

Mostramos primeras columnas donde se observan valores 'NaN'

In [31]:
tortilla.head(3)

State            City  Year  Month  Day         Store type  \
0   Aguascalientes  Aguascalientes  2007      1   10  Mom and Pop Store   
1  Baja California        Mexicali  2007      1   10  Mom and Pop Store   
2  Baja California         Tijuana  2007      1   10  Mom and Pop Store   

   Price per kilogram  
0                 9.9  
1                 NaN  
2                10.0

Mostramos limpieza de datos en primeras columnas

In [30]:
tortilla_clean.head(3)

State            City  Year  Month  Day         Store type  \
0       Aguascalientes  Aguascalientes  2007      1   10  Mom and Pop Store   
2      Baja California         Tijuana  2007      1   10  Mom and Pop Store   
3  Baja California Sur          La Paz  2007      1   10  Mom and Pop Store   

   Price per kilogram  
0                 9.9  
2                10.0  
3                10.0

In [32]:
tortilla_clean.describe()

Year          Month            Day  Price per kilogram
count  278004.000000  278004.000000  278004.000000       278004.000000
mean     2015.196148       6.349142      15.918814           12.203067
std         5.027872       3.431041       8.751713            4.490489
min      2007.000000       1.000000       1.000000            0.000000
25%      2011.000000       3.000000       8.000000            9.600000
50%      2015.000000       6.000000      16.000000           11.230000
75%      2019.000000       9.000000      24.000000           14.090000
max      2024.000000      12.000000      31.000000           31.000000

- El precio medio del kilogramo de tortilla en México de 2007 a 2024 es de $12.20


In [36]:
# Mostramos precios por kiogramo mayores a la media de 2007 2024
precios_altos = tortilla_clean [tortilla_clean['Price per kilogram'] > 12.203067]


In [37]:
precios_altos.describe()

Year          Month            Day  Price per kilogram
count  113143.000000  113143.000000  113143.000000       113143.000000
mean     2018.855334       6.451499      15.870854           16.322866
std         3.878479       3.433018       8.754115            3.947896
min      2008.000000       1.000000       1.000000           12.210000
25%      2016.000000       4.000000       8.000000           13.600000
50%      2020.000000       6.000000      16.000000           15.000000
75%      2022.000000       9.000000      23.500000           17.880000
max      2024.000000      12.000000      31.000000           31.000000

In [41]:
precios_altos_max = precios_altos[precios_altos['Price per kilogram'] == 31]
precios_altos_max

State        City  Year  Month  Day         Store type  \
276230  Tamaulipas   Matamoros  2024      1    3  Mom and Pop Store   
276284  Tamaulipas   Matamoros  2024      1    5  Mom and Pop Store   
276338  Tamaulipas   Matamoros  2024      1    8  Mom and Pop Store   
276392  Tamaulipas   Matamoros  2024      1   10  Mom and Pop Store   
276446  Tamaulipas   Matamoros  2024      1   12  Mom and Pop Store   
...            ...         ...   ...    ...  ...                ...   
280172  Tamaulipas   Matamoros  2024      7    3  Mom and Pop Store   
280221      Sonora  Hermosillo  2024      7    5  Mom and Pop Store   
280226  Tamaulipas   Matamoros  2024      7    5  Mom and Pop Store   
280275      Sonora  Hermosillo  2024      7    8  Mom and Pop Store   
280280  Tamaulipas   Matamoros  2024      7    8  Mom and Pop Store   

        Price per kilogram  
276230                31.0  
276284                31.0  
276338                31.0  
276392                31.0  
276446                31.0  
...                    ...  
280172                31.0  
280221                31.0  
280226                31.0  
280275                31.0  
280280                31.0  

[128 rows x 7 columns]

El precio mas alto en México de 2007 al 2024 fué de $31.00 en Matamoros Tamaulipas y Hermosillo Sonora en el año 2024 

In [60]:
# Precio max especifico en el estado de Tlaxcala
texcoco_tlaxcala = tortilla_clean[tortilla_clean['State'] == 'Tlaxcala']
texcoco_tlaxcala.describe()


Year        Month          Day  Price per kilogram
count  5212.000000  5212.000000  5212.000000         5212.000000
mean   2015.189179     6.341136    15.929010           10.005627
std       5.075811     3.431582     8.750488            2.621900
min    2007.000000     1.000000     1.000000            4.470000
25%    2011.000000     3.000000     8.000000            8.000000
50%    2015.000000     6.000000    16.000000           10.000000
75%    2020.000000     9.000000    24.000000           11.000000
max    2024.000000    12.000000    31.000000           18.000000

In [61]:
texcoco_tlaxcala_max = texcoco_tlaxcala[texcoco_tlaxcala['Price per kilogram'] == 18]
texcoco_tlaxcala_max.head(3)

State      City  Year  Month  Day         Store type  \
278556  Tlaxcala  Tlaxcala  2024      4   19  Mom and Pop Store   
278610  Tlaxcala  Tlaxcala  2024      4   22  Mom and Pop Store   
278664  Tlaxcala  Tlaxcala  2024      4   24  Mom and Pop Store   

        Price per kilogram  
278556                18.0  
278610                18.0  
278664                18.0

El precio más alto en el municipio de Tlaxcala, Tlaxcala de 2007 al 2024 fué de $18.00 en 2024

In [76]:
# Mostramos el estado con el precio mas bajo en México

estado_mas_bajo = tortilla_clean[tortilla_clean['Year'] == 2024]


In [67]:
estado_mas_bajo.describe()

Year        Month         Day  Price per kilogram
count  8206.0  8206.000000  8206.00000         8206.000000
mean   2024.0     3.684865    14.97319           19.508627
std       0.0     1.850970     8.56393            5.691428
min    2024.0     1.000000     1.00000           11.600000
25%    2024.0     2.000000     8.00000           14.130000
50%    2024.0     4.000000    15.00000           17.900000
75%    2024.0     5.000000    22.00000           24.737500
max    2024.0     7.000000    31.00000           31.000000

In [81]:
estado_mas_bajo_min = estado_mas_bajo[estado_mas_bajo['Price per kilogram'] == 11.6]
estado_mas_bajo_min.head()


State           City  Year  Month  Day        Store type  \
280393  Veracruz  Coatzacoalcos  2024      1    5  Big Retail Store   
280447  Veracruz  Coatzacoalcos  2024      1    8  Big Retail Store   

        Price per kilogram  
280393                11.6  
280447                11.6

In [87]:
print(f'El precio mas bajo en 2024 en México fue en: {estado_mas_bajo_min}')


El precio mas bajo en 2024 en México fue en:            State           City  Year  Month  Day        Store type  \
280393  Veracruz  Coatzacoalcos  2024      1    5  Big Retail Store   
280447  Veracruz  Coatzacoalcos  2024      1    8  Big Retail Store   

        Price per kilogram  
280393                11.6  
280447                11.6  
